src/components/global/BaseButton.vue

```diff
@@ -69,6 +69,7 @@ export default defineComponent({
       :class="buttonClasses"
       :colorScheme="colorScheme"
       @click="$emit('click', $event)"
+      @submit="$emit('submit', $event)"
```

Comment:
How come our button has a `submit` event? 😅
Forms do but... button? 🤔

Reply:
Oh, here I've really overdone it..
TBH, this PR got a bit bigger then I initially intended (definitely not proud of that) and as You can see I got mixed up in a couple of places.
Of course I already submit the data via the parent form component.